src/main/java/io/specto/hoverfly/junit/rule/NoDiffAssertionRule.java

```diff
@@ -0,0 +1,100 @@
+ */
+public class NoDiffAssertionRule implements TestRule {
+
+    private Hoverfly hoverfly;
+    private final boolean shouldClean;
+    private HoverflyRule hoverflyRule;
+
+    /**
+     * Creates a rule with the given instance of {@link Hoverfly} that asserts that there was no diff between any of the
+     * expected responses set by simulations and the actual responses returned from the real service.
+     * The parameter {@code shouldClean} says if the rule should clean all available diffs stored in Hoverfly before the
+     * test execution and when the assertion is performed (after the test method execution).
+     *
+     * @param hoverfly An instance of {@link Hoverfly} to be used for retrieving the diffs
+     * @param shouldClean if all available diffs should be removed.
+     */
+    public NoDiffAssertionRule(Hoverfly hoverfly, boolean shouldClean) {
+        if (hoverfly == null) {
+            throw new IllegalArgumentException("Hoverfly cannot be null");
+        }
+        this.hoverfly = hoverfly;
+        this.shouldClean = shouldClean;
+    }
+
+    /**
+     * Creates a rule with the given instance of {@link HoverflyRule} that asserts that there was no diff between any of the
+     * expected responses set by simulations and the actual responses returned from the real service.
+     * The parameter {@code shouldClean} says if the rule should clean all available diffs stored in Hoverfly before the
+     * test execution and when the assertion is performed (after the test method execution).
+     *
+     * @param hoverflyRule An instance of {@link HoverflyRule} to be used for retrieving the diffs
+     * @param shouldClean if all available diffs should be removed.
+     */
+    public NoDiffAssertionRule(HoverflyRule hoverflyRule, boolean shouldClean) {
+        if (hoverflyRule == null) {
+            throw new IllegalArgumentException("HoverflyRule cannot be null");
+        }
+        this.hoverflyRule = hoverflyRule;
+        this.shouldClean = shouldClean;
+    }
+
+    /**
+     * Creates a rule with the given instance of {@link HoverflyConfig} that asserts that there was no diff between any of the
+     * expected responses set by simulations and the actual responses returned from the real service.
+     * The parameter {@code shouldClean} says if the rule should clean all available diffs stored in Hoverfly before the
+     * test execution and when the assertion is performed (after the test method execution).
+     *
+     * @param hoverflyConfig An instance of {@link HoverflyConfig} to be used for retrieving the diffs
+     * @param shouldClean if all available diffs should be removed.
+     */
+    public NoDiffAssertionRule(HoverflyConfig hoverflyConfig, boolean shouldClean) {
+        if (hoverflyConfig == null) {
+            throw new IllegalArgumentException("HoverflyConfig cannot be null");
+        }
+        hoverfly = new Hoverfly(hoverflyConfig, HoverflyMode.DIFF);
+        this.shouldClean = shouldClean;
```

Comment:
Do we really need this additional option to reset the diff or not during each test run? It's very likely that it could be always true, so maybe we can be a bit opinionated and better off dropping this boolean field, like you did for `hoverflyRule. assertThatNoDiffIsReported()`

Reply:
Do you mean to remove that from the `NoDiffAssertionRule` completely, or to just add other constructors without the param that would take the default value: `true`?

Reply:
I think you can remove it unless there is a use case for not resetting diff between each test.

Reply:
done in 43b731d